### ISSUE TITLE
Review database design

### ISSUE THREAD

**Issue by J-Mead** [SeattleDataforGood/Lead_Water_Project#13]

**Comment by Phinneas** [comment]
1) Do we need the StateContaminantThreshold table and the SchoolDistrictContaminantThreshold tables? Wouldn't the threshold be the same for both?
2)Facility table, should we make FacilityAddressLine1, FacilityCity, FacilityStateCode, and FacilityPostalCode be enforced as 'not null'?
3)If we have a SchoolDistrictUrl table do we need a DistrictURL column within the SchoolDistrict table?

**Comment by J-Mead** [comment]
For Buzz's question 1: the reason we have StateContaminantThreshold and SchoolDistrictContaminantThreshold is that there is already a school district with a different threshold from the state in lead: Tacoma. This came up in the last meeting. It's probably not very common, but since we already know that a district is doing this we didn't want to pretend the situation didn't exist.

**Comment by J-Mead** [comment]
I was thinking about the moving schools, and I think we can handle it by adding a column in the school table for active/inactive/closed. We only report on active schools. If a school moves to a temporary location, then the clients can create a "new" school there, and make it inactive when the school moves back to its original location. That way, the clients take on the hassle of moving schools and we do not have to maintain history. Maybe we give them one report with inactive and closed schools too. What do you think of that?

Buzz Beard [5:38 PM] 
@jennifermead  @mariehaggberg I think Jen your solution makes some sense to me because if a school changes location the old location tests would not be relevant. Unless there would be a case where the school returned to the old location, or some other state entity moved into that location. In that case it might be nice to maintain that data somewhere. Maybe thats a feature that may be beyond our current scope.

Jen Mead [6:42 PM] 
The old data would still be there under the name of the original location, marked inactive until the space was used again.

[6:46] 
School A is no longer used and marked inactive. School B needs remodeling so they temporarily move to the location of School A. School B is marked inactive. The clients create a "new" school in the database called "School B at Old School A" and copy the fixture data from A into the temp record, marking it active. When the remodel is complete, "School B at Old School A" is marked inactive. The original record for School B is marked active again, with test results for new fixtures added in the remodel.

[6:48] 
Basically the clients create/mark active/mark inactive however they want. As far as our database is concerned, it's just another school. The name or address of the school doesn't matter to us. The clients mark active whichever set of records they want the reports and maps to show.

**Comment by J-Mead** [comment]
Other design suggestions:
--Client would like a column for WaterTestable with yes/no values
--We should add a column for active/inactive/demolished for schools. This will also be useful in the future when we add asbestos testing, because they track whether schools have been demolished.
-- Can we add the name of the lab that performs a test? This would allow us to analyze in the future if one lab finds more or fewer issues. 

An alternate way to handle moving schools might be to have the school name and number in a separate table than the school address, then have dates for when the school is at this address. I'm not sure this will work for all cases. What do you think? Constraints/facts include:
-- Schools can move to a temporary location (ex: during remodel)
-- Schools can move to a new permanent location (ex: old location found unsafe or too small)
-- Schools can change names (ex: named after recently deceased politician)
-- More than one school can occupy the same temporary location
-- Schools can add and remove fixtures during remodels
-- Schools can be demolished (and fixtures "removed")

As far as frequency, most schools in the state will not move. However, in Seattle there are currently two or three schools in temp locations, and it is a regular thing for this to happen. I suspect it is more common in highly population dense districts, because the school grounds are limited by other buildings.

**Comment by hcsea** [comment]
Hi all, currently reviewing items here - busy couple of weeks with a coworker out sick. Re: school locations, Jen's suggestion "have the school name and number in a separate table than the school address, then have dates for when the school is at this address" is IMHO a more correct design approach. I can proceed on this and other requests this weekend.

**Comment by hcsea** [comment]
On Buzz's comment re: the contaminant tables, I can set a default for Last Updated to be current date and time, and make it not null. Re: "allowable per" I'm unsure if we'd have cases where no amount is allowable, e.g. if there is substance X in water or air, test fails, regardless of measure. Thoughts?
